symengine/functions.cpp

```diff
@@ -3694,4 +3694,26 @@ RCP<const Basic> min(const vec_basic &arg)
     }
 }
 
+UnevaluatedExpr::UnevaluatedExpr(const RCP<const Basic> &arg)
+    : OneArgFunction(arg)
+{
+    SYMENGINE_ASSIGN_TYPEID()
+    SYMENGINE_ASSERT(is_canonical(arg))
+}
+
+bool UnevaluatedExpr::is_canonical(const RCP<const Basic> &arg) const
+{
+    return true;
+}
```

Comment:
I think the constructor should go into the header file, so that the compiler can inline it. I know we have been putting them into the cpp files, but I think that was a mistake.

Reply:
(We can fix this for the whole SymEngine in a separate PR.)

Reply:
True, but except for a few types like Integer, UnevaluatedExpr, we don't usually call the constructor in downstream code